WIP: Update msvc build to ignore warning and bump Qt version to 5.12.6.

sipsorcery opened the pull request:
WIP while checking whether the Bitcoin Core appveyor job passes. This depends on the state of the vcpkg packages which can't be replicated on a fork.

msvc warning C4834 for the Bitcoin Core build was introduced by Visual Studio 16.4.0. This PR adds an ignore rule for the warning (it's related to the nodiscard attribute and is not considered relevant).

An additional side effect of the msvc compiler update is the cached Qt5.9.8 libraries cannot be linked due to being built with an earlier version of the compiler. To fix this Qt5.12.6 has been built with the updated msvc compiler and added to the appveyor job.

sipsorcery committed:
Update msvc build to ignore warning and bump Qt version to 5.12.6.

msvc warning C4834 for the Bitcoin Core build was introduced by Visual Studio 16.4.0. This PR adds an ignore rule for the warning (it's related to the nodiscard attribute and is not considered relevant).
An additional side effect of the msvc compiler update is the cached Qt5.9.8 libraries cannot be linked due to being built with an earlier version of the compiler. To fix this Qt5.12.6 has been built with the updated msvc compiler and added to the appveyor job.

fanquake commented on build_msvc/common.init.vcxproj:
```diff
@@ -107,15 +107,15 @@
       <SubSystem>Console</SubSystem>
       <GenerateDebugInformation>true</GenerateDebugInformation>
-      <AdditionalDependencies>crypt32.lib;Iphlpapi.lib;ws2_32.lib;Shlwapi.lib;kernel32.lib;user32.lib;gdi32.lib;winspool.lib;comdlg32.lib;advapi32.lib;shell32.lib;ole32.lib;oleaut32.lib;uuid.lib;odbc32.lib;odbccp32.lib;%(AdditionalDependencies)</AdditionalDependencies>
+      <AdditionalDependencies>Wtsapi32.lib;crypt32.lib;Iphlpapi.lib;ws2_32.lib;Shlwapi.lib;kernel32.lib;user32.lib;gdi32.lib;winspool.lib;comdlg32.lib;advapi32.lib;shell32.lib;ole32.lib;oleaut32.lib;uuid.lib;odbc32.lib;odbccp32.lib;%(AdditionalDependencies)</AdditionalDependencies>
```
> Wtsapi32.lib

Why is this needed? A quick search suggests it is the library for the `Remote Desktop Services API`.

sipsorcery replied:
Qt5.12.6 wanted it. That did raise an orange flag for me as well. Some quick research suggested it's to allow Qt apps to be served across remote terminal connections rather than any kind of hosting ability.

sipsorcery commented:
When the `Wstapi32.lib` is not referenced the linker error is:

````
"c:\Dev\github\sipsorcery_bitcoin\build_msvc\bitcoin.sln" (build target) (1) ->
"c:\Dev\github\sipsorcery_bitcoin\build_msvc\bitcoin-qt\bitcoin-qt.vcxproj" (default target) (23) ->
(Link target) ->
 qwindows.lib(qwindowscontext.obj) : error LNK2001: unresolved external symbol WTSQuerySessionInformationW [c:\
Dev\github\sipsorcery_bitcoin\build_msvc\bitcoin-qt\bitcoin-qt.vcxproj]
 qwindows.lib(qwindowscontext.obj) : error LNK2001: unresolved external symbol WTSFreeMemory [c:\Dev\github\sip
sorcery_bitcoin\build_msvc\bitcoin-qt\bitcoin-qt.vcxproj]
 c:\Dev\github\sipsorcery_bitcoin\build_msvc\x64\Release\bitcoin-qt.exe : fatal error LNK1120: 2 unresolved ext
ernals [c:\Dev\github\sipsorcery_bitcoin\build_msvc\bitcoin-qt\bitcoin-qt.vcxproj]
````

The two Remote Desktop functions being used are [WTSQuerySessionInformationW](https://docs.microsoft.com/en-us/windows/win32/api/wtsapi32/nf-wtsapi32-wtsquerysessioninformationw) and [WTSFreeMemory](https://docs.microsoft.com/en-us/windows/win32/api/wtsapi32/nf-wtsapi32-wtsfreememory).

A query and free memory call so no obvious capabilities to start a remote session or such.

I am working on a new build of Qt incorporating #17730 to see if the need for `Wstapi32.lib` can be eliminated. My gut feel is it can't since it's being called from the Qt `qwindows.lib` and based on Qt's `configure -list-features` and `configure -list-libraries` there are no obvious options related to it.

sipsorcery committed:
Set the CI Actions workflow to use the Qt5.9.8 path for msvc build.

The Qt5.12.6 binaries have been compiled with a newer compiler than is available on the Actions Windows image. This fix allows the Actions workflow to override the Qt path in the common project file by setting it on the msbuild command line.

maflcko reviewed (approved):
ACK. Didn't look at the changes, but this makes appveyor green.

laanwj commented:
> A query and free memory call so no obvious capabilities to start a remote session or such.

Ok, it's kind of worrying that Qt keeps adding these kind of requirements, but not necessary to handle here.

ACK if Appveyor passes again.

sipsorcery commented:
To make this less contentious I'll rebuild Qt5.9.8 with Visual Studio 2019 v16.4 and resubmit the PR without needing to bump Qt.
